pages/example-3-missing-authorization-api-route/vulnerable-2.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { BlogPost } from '../../database/blogPosts';
+import Common from './common';
+import { User } from '../../database/users';
+
+type Props = {
+  userObject: User;
+};
+
+export default function MissingAuthentication(props: Props) {
+  const [error, setError] = useState<string>();
+  const [blogPosts, setBlogPosts] = useState<BlogPost[]>([]);
+
+  useEffect(() => {
+    async function fetchInitialData() {
+      const response = await fetch(
+        '/api/example-3-missing-authorization-api-route/vulnerable',
+      );
+
+      const data = await response.json();
+
+      if ('error' in data) {
+        setError(data.error);
+        return;
+      }
+
+      setError(undefined);
+
+      setBlogPosts([]);
+      data.blogPosts.map(blogPost => {
```

Comment:
I guess here it would be simpler and easier to follow with this pattern:

```js
setBlogPosts(data.blogPosts.filter( ... ))
```

Then you also only have a single call to setBlogPosts, which improves performance

Reply:
uh, fancy - like that =)